src/modules/aws_cloudfront/index.ts

```diff
@@ -117,17 +120,30 @@ class DistributionMapper extends MapperBase<Distribution> {
       WebACLId: e.webACLId,
       Origins: { Quantity: e.origins.length, Items: e.origins },
       PriceClass: distributionConfig.DistributionConfig?.PriceClass,
-      Aliases: distributionConfig.DistributionConfig?.Aliases,
+      Aliases: { Quantity: e.alternateDomainNames?.length ?? 0, Items: e.alternateDomainNames ?? [] },
       Logging: distributionConfig.DistributionConfig?.Logging,
       CacheBehaviors: distributionConfig.DistributionConfig?.CacheBehaviors,
       CustomErrorResponses: distributionConfig.DistributionConfig?.CustomErrorResponses,
       DefaultRootObject: distributionConfig.DistributionConfig?.DefaultRootObject,
       HttpVersion: distributionConfig.DistributionConfig?.HttpVersion,
       OriginGroups: distributionConfig.DistributionConfig?.OriginGroups,
       Restrictions: distributionConfig.DistributionConfig?.Restrictions,
-      ViewerCertificate: distributionConfig.DistributionConfig?.ViewerCertificate,
     };
 
+    if (e.customSslCertificate?.arn)
+      req.ViewerCertificate = {
+        CloudFrontDefaultCertificate: false,
+        ACMCertificateArn: e.customSslCertificate.arn,
+        MinimumProtocolVersion: MinimumProtocolVersion.TLSv1_2_2021,
```

Comment:
Will we expose these parameters later, or is this one of those situations where you have to provide a specific value or it fails and the argument itself is deprecated?

Reply:
Yes. I think we should expose those values later, but since they're rarely used I've not added support for those yet.